ref: https://github.com/FreeCAD/FreeCAD/commit/018b41032a5af616fb861538695f8be14ad120ce
ArchLinux and OpenSuse also use it.
Adapted to support patchlevel #0 and dos2unix

commit 018b41032a5af616fb861538695f8be14ad120ce
Author: sL1pKn07 <sl1pkn07@gmail.com>
Date:   Sat Apr 7 02:17:49 2018 +0200

    [For v018] Fix build with Qt5 beta
    
    Seems need add some headers for build with incoming Qt 5.11 (Tested with 5.11.0beta2)

diff --git a/src/Mod/Fem/Gui/TaskFemConstraintBearing.cpp b/src/Mod/Fem/Gui/TaskFemConstraintBearing.cpp
index c3992f9d8..cd9b64a94 100644
--- src/Mod/Fem/Gui/TaskFemConstraintBearing.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintBearing.cpp
@@ -28,6 +28,7 @@
 # include <QRegExp>
 # include <QTextStream>
 # include <QMessageBox>
+# include <QAction>
 # include <Precision.hxx>
 # include <TopoDS.hxx>
 # include <BRepAdaptor_Surface.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintContact.cpp b/src/Mod/Fem/Gui/TaskFemConstraintContact.cpp
index c445ca1c8..6fe9898e1 100644
--- src/Mod/Fem/Gui/TaskFemConstraintContact.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintContact.cpp
@@ -32,6 +32,7 @@
 # include <Geom_Plane.hxx>
 # include <Precision.hxx>
 # include <QMessageBox>
+# include <QAction>
 # include <QRegExp>
 # include <QTextStream>
 # include <TopoDS.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintDisplacement.cpp b/src/Mod/Fem/Gui/TaskFemConstraintDisplacement.cpp
index 0a83e737b..cfb16d9d8 100644
--- src/Mod/Fem/Gui/TaskFemConstraintDisplacement.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintDisplacement.cpp
@@ -32,6 +32,7 @@
 # include <Geom_Plane.hxx>
 # include <Precision.hxx>
 # include <QMessageBox>
+# include <QAction>
 # include <QRegExp>
 # include <QTextStream>
 # include <TopoDS.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintFixed.cpp b/src/Mod/Fem/Gui/TaskFemConstraintFixed.cpp
index 0d864e47e..183719432 100644
--- src/Mod/Fem/Gui/TaskFemConstraintFixed.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintFixed.cpp
@@ -29,6 +29,7 @@
 # include <Geom_Plane.hxx>
 # include <Precision.hxx>
 # include <QMessageBox>
+# include <QAction>
 # include <QRegExp>
 # include <QTextStream>
 # include <TopoDS.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintFluidBoundary.cpp b/src/Mod/Fem/Gui/TaskFemConstraintFluidBoundary.cpp
index c9964403b..8aaacdcf4 100644
--- src/Mod/Fem/Gui/TaskFemConstraintFluidBoundary.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintFluidBoundary.cpp
@@ -29,6 +29,7 @@
 # include <QRegExp>
 # include <QTextStream>
 # include <QMessageBox>
+# include <QAction>
 # include <Precision.hxx>
 # include <TopoDS.hxx>
 # include <TopoDS_Shape.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintForce.cpp b/src/Mod/Fem/Gui/TaskFemConstraintForce.cpp
index 33deb1b00..94fa4209a 100644
--- src/Mod/Fem/Gui/TaskFemConstraintForce.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintForce.cpp
@@ -28,6 +28,7 @@
 # include <QRegExp>
 # include <QTextStream>
 # include <QMessageBox>
+# include <QAction>
 # include <Precision.hxx>
 # include <TopoDS.hxx>
 # include <BRepAdaptor_Surface.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintHeatflux.cpp b/src/Mod/Fem/Gui/TaskFemConstraintHeatflux.cpp
index 344b8d32b..20a636215 100644
--- src/Mod/Fem/Gui/TaskFemConstraintHeatflux.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintHeatflux.cpp
@@ -32,6 +32,7 @@
 # include <Geom_Plane.hxx>
 # include <Precision.hxx>
 # include <QMessageBox>
+# include <QAction>
 # include <QRegExp>
 # include <QTextStream>
 # include <TopoDS.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintPlaneRotation.cpp b/src/Mod/Fem/Gui/TaskFemConstraintPlaneRotation.cpp
index b4c3cb03b..29d90c334 100644
--- src/Mod/Fem/Gui/TaskFemConstraintPlaneRotation.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintPlaneRotation.cpp
@@ -30,6 +30,7 @@
 # include <QRegExp>
 # include <QTextStream>
 # include <QMessageBox>
+# include <QAction>
 # include <Precision.hxx>
 # include <TopoDS.hxx>
 # include <BRepAdaptor_Surface.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintPressure.cpp b/src/Mod/Fem/Gui/TaskFemConstraintPressure.cpp
index 2da2b11cb..a85289183 100644
--- src/Mod/Fem/Gui/TaskFemConstraintPressure.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintPressure.cpp
@@ -30,6 +30,7 @@
 # include <Geom_Plane.hxx>
 # include <Precision.hxx>
 # include <QMessageBox>
+# include <QAction>
 # include <QRegExp>
 # include <QTextStream>
 # include <TopoDS.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintTemperature.cpp b/src/Mod/Fem/Gui/TaskFemConstraintTemperature.cpp
index 42c0d8f2e..88e9029fd 100644
--- src/Mod/Fem/Gui/TaskFemConstraintTemperature.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintTemperature.cpp
@@ -32,6 +32,7 @@
 # include <Geom_Plane.hxx>
 # include <Precision.hxx>
 # include <QMessageBox>
+# include <QAction>
 # include <QRegExp>
 # include <QTextStream>
 # include <TopoDS.hxx>
diff --git a/src/Mod/Fem/Gui/TaskFemConstraintTransform.cpp b/src/Mod/Fem/Gui/TaskFemConstraintTransform.cpp
index 35a1fcf5b..e3afeb785 100644
--- src/Mod/Fem/Gui/TaskFemConstraintTransform.cpp
+++ src/Mod/Fem/Gui/TaskFemConstraintTransform.cpp
@@ -33,6 +33,7 @@
 # include <Geom_Plane.hxx>
 # include <Precision.hxx>
 # include <QMessageBox>
+# include <QAction>
 # include <QRegExp>
 # include <QTextStream>
 # include <TopoDS.hxx>
diff --git a/src/Mod/MeshPart/Gui/Tessellation.cpp b/src/Mod/MeshPart/Gui/Tessellation.cpp
index e392d470b..7636ad2fb 100644
--- src/Mod/MeshPart/Gui/Tessellation.cpp
+++ src/Mod/MeshPart/Gui/Tessellation.cpp
@@ -25,6 +25,7 @@
 #ifndef _PreComp_
 # include <TopExp_Explorer.hxx>
 # include <QMessageBox>
+# include <QButtonGroup>
 #endif
 
 #include "Tessellation.h"
diff --git a/src/Mod/Part/Gui/DlgSettingsGeneral.cpp b/src/Mod/Part/Gui/DlgSettingsGeneral.cpp
index d0cf0f94e..b44eac6ff 100644
--- src/Mod/Part/Gui/DlgSettingsGeneral.cpp
+++ src/Mod/Part/Gui/DlgSettingsGeneral.cpp
@@ -22,6 +22,9 @@
 
 
 #include "PreCompiled.h"
+#ifndef _PreComp_
+# include <QButtonGroup>
+#endif
 
 #include <Interface_Static.hxx>
 
diff --git a/src/Mod/PartDesign/Gui/TaskBooleanParameters.cpp b/src/Mod/PartDesign/Gui/TaskBooleanParameters.cpp
index 785705448..7e3280065 100644
--- src/Mod/PartDesign/Gui/TaskBooleanParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskBooleanParameters.cpp
@@ -25,6 +25,7 @@
 
 #ifndef _PreComp_
 # include <QMessageBox>
+# include <QAction>
 #endif
 
 #include "ui_TaskBooleanParameters.h"
diff --git a/src/Mod/PartDesign/Gui/TaskChamferParameters.cpp b/src/Mod/PartDesign/Gui/TaskChamferParameters.cpp
index 330d58535..4600cc25b 100644
--- src/Mod/PartDesign/Gui/TaskChamferParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskChamferParameters.cpp
@@ -24,6 +24,7 @@
 #include "PreCompiled.h"
 
 #ifndef _PreComp_
+# include <QAction>
 #endif
 
 #include "ui_TaskChamferParameters.h"
diff --git a/src/Mod/PartDesign/Gui/TaskDraftParameters.cpp b/src/Mod/PartDesign/Gui/TaskDraftParameters.cpp
index 556fc92ae..020148001 100644
--- src/Mod/PartDesign/Gui/TaskDraftParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskDraftParameters.cpp
@@ -25,6 +25,7 @@
 
 #ifndef _PreComp_
 # include <QMessageBox>
+# include <QAction>
 #endif
 
 #include "ui_TaskDraftParameters.h"
diff --git a/src/Mod/PartDesign/Gui/TaskFilletParameters.cpp b/src/Mod/PartDesign/Gui/TaskFilletParameters.cpp
index 3c4025574..5e0c31ec6 100644
--- src/Mod/PartDesign/Gui/TaskFilletParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskFilletParameters.cpp
@@ -24,6 +24,7 @@
 #include "PreCompiled.h"
 
 #ifndef _PreComp_
+# include <QAction>
 #endif
 
 #include "ui_TaskFilletParameters.h"
diff --git a/src/Mod/PartDesign/Gui/TaskLinearPatternParameters.cpp b/src/Mod/PartDesign/Gui/TaskLinearPatternParameters.cpp
index 532936310..1970db7ef 100644
--- src/Mod/PartDesign/Gui/TaskLinearPatternParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskLinearPatternParameters.cpp
@@ -25,6 +25,7 @@
 
 #ifndef _PreComp_
 # include <QMessageBox>
+# include <QAction>
 # include <QTimer>
 #endif
 
diff --git a/src/Mod/PartDesign/Gui/TaskMirroredParameters.cpp b/src/Mod/PartDesign/Gui/TaskMirroredParameters.cpp
index 084ab353c..f8e96fd6c 100644
--- src/Mod/PartDesign/Gui/TaskMirroredParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskMirroredParameters.cpp
@@ -25,6 +25,7 @@
 
 #ifndef _PreComp_
 # include <QMessageBox>
+# include <QAction>
 #endif
 
 #include <Base/Console.h>
diff --git a/src/Mod/PartDesign/Gui/TaskMultiTransformParameters.cpp b/src/Mod/PartDesign/Gui/TaskMultiTransformParameters.cpp
index 1fe452a59..d5a5317ae 100644
--- src/Mod/PartDesign/Gui/TaskMultiTransformParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskMultiTransformParameters.cpp
@@ -25,6 +25,7 @@
 
 #ifndef _PreComp_
 # include <QMessageBox>
+# include <QAction>
 #endif
 
 #include "ui_TaskMultiTransformParameters.h"
diff --git a/src/Mod/PartDesign/Gui/TaskPolarPatternParameters.cpp b/src/Mod/PartDesign/Gui/TaskPolarPatternParameters.cpp
index 874273f63..79c564d09 100644
--- src/Mod/PartDesign/Gui/TaskPolarPatternParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskPolarPatternParameters.cpp
@@ -25,6 +25,7 @@
 
 #ifndef _PreComp_
 # include <QMessageBox>
+# include <QAction>
 # include <QTimer>
 #endif
 
diff --git a/src/Mod/PartDesign/Gui/TaskScaledParameters.cpp b/src/Mod/PartDesign/Gui/TaskScaledParameters.cpp
index 593664c9b..acaacce6e 100644
--- src/Mod/PartDesign/Gui/TaskScaledParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskScaledParameters.cpp
@@ -25,6 +25,7 @@
 
 #ifndef _PreComp_
 # include <QMessageBox>
+# include <QAction>
 #endif
 
 #include "ui_TaskScaledParameters.h"
diff --git a/src/Mod/PartDesign/Gui/TaskThicknessParameters.cpp b/src/Mod/PartDesign/Gui/TaskThicknessParameters.cpp
index b3f17f299..5b907304f 100644
--- src/Mod/PartDesign/Gui/TaskThicknessParameters.cpp
+++ src/Mod/PartDesign/Gui/TaskThicknessParameters.cpp
@@ -24,6 +24,7 @@
 #include "PreCompiled.h"
 
 #ifndef _PreComp_
+# include <QAction>
 #endif
 
 #include "ui_TaskThicknessParameters.h"
